server/test/api/purchase.js

```diff
@@ -0,0 +1,151 @@
+      total : 10.7520 
+    },
+    { 
+      uuid : uuid(),       
+      inventory_uuid : 'c48a3c4b-c07d-4899-95af-411f7708e296',
+      quantity : 16000,
+      unit_price : 0.0335,
+      total : 536.0000 
+    }    
+   ]; 
+
+  var purchase = {
+    purchase_order : purchase_order,
+    purchase_item : purchase_item
+  };
+
+  var invalidPurchase = {
+    inv_purchase_order : {},
+    inv_purchase_item : {}
+  };
+
+
+  // login before each request  
+  beforeEach(helpers.login(agent));
+
+  it('POST /purchase should create a new purchase order', function () {
+
+    return agent.post('/purchase')
+      .send(purchase)
+      .then(function (res) {
+        expect(res).to.have.status(201);
+        expect(res).to.be.json;
+        var purchaseUuid = res.body.uuid;
+        return agent.get('/purchase/' + purchaseUuid);
+      })
+      .then(function (res) {
+        var purchase = res.body;
+
+        expect(res).to.have.status(200);
+        expect(res).to.be.json;
+        expect(purchase.cost).to.equal(purchase_order.cost);
+      })
+      .catch(helpers.handler);
+  });
+
+
+  it('GET /purchase/:uuid should return a single JSON purchase order', function () {
+    return agent.get('/purchase/' + purchase_order.uuid)
+      .then(function (res) {
+        var purchase = res.body.purchase;
+
+        expect(res).to.have.status(200);
+        expect(res.body).to.not.be.empty;
+        expect(purchase.uuid).to.exist;
+      })
+      .catch(helpers.handler);
+  });
+
+  it('GET /purchase/ ? COMPLETE = 1 returns a complete List of purchase   ', function () { 
+    return agent.get('/purchase?complete=1')
+      .then(function (result) {
+        expect(result).to.have.status(200);
+        expect(result).to.be.json;
+        expect(result.body[0]).to.contain.all.keys(PURCHASE_KEY); 
+      })
+      .catch(helpers.handler);
+  });
+
+  it('GET /purchase/:uuid returns 404 for an invalid purchase order', function () {
+    return agent.get('/purchase/unkown')
+      .then(function (result) {
+        expect(result).to.have.status(404);
+        expect(result.body).to.not.be.empty;
+        expect(result.body.code).to.equal('ERR_NOT_FOUND');
+      })
+      .catch(helpers.handler);
+  });
+
+  it('PUT /purchase should update an property of a Purchase Order ', function () {
+    return agent.put('/purchase/' + purchase_order.uuid)
+      .send({ is_validate : 1 })
+      .then(function (res) {
+        expect(res).to.have.status(200);
+        expect(res.body.is_validate).to.not.equal(purchase_order.is_validate);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('POST /purchase returns 400 for an invalid Purchase Order request object', function () {
+
+    return agent.post('/purchase')
+      .send(invalidPurchase)
+      .then(function (res) {
+        expect(res).to.have.status(400);
+        expect(res.body).to.not.be.empty;
+      });
+  });
+
+  it('PUT /purchase should update an Invalid Purchase Order ', function () {
```

Comment:
This says the test **should** update an invalid purchase order and then checks for a 500 error.  Either the label text is wrong or the test is wrong.